File: src/FantasyCritic.Lib/Royale/RoyalePublisher.cs
using FantasyCritic.Lib.Identity;

namespace FantasyCritic.Lib.Royale;

public class RoyalePublisher
{
    public RoyalePublisher(Guid publisherID, RoyaleYearQuarter yearQuarter, VeryMinimalFantasyCriticUser user,
        string publisherName, string? publisherIcon, string? publisherSlogan, IEnumerable<RoyalePublisherGame> publisherGames, decimal budget)
    {
        PublisherID = publisherID;
        YearQuarter = yearQuarter;
        User = user;
        PublisherName = publisherName;
        PublisherIcon = publisherIcon;
        PublisherSlogan = publisherSlogan;
        PublisherGames = publisherGames.ToList();
        Budget = budget;
    }

    public Guid PublisherID { get; }
    public RoyaleYearQuarter YearQuarter { get; }
    public VeryMinimalFantasyCriticUser User { get; }
    public string PublisherName { get; }
    public string? PublisherIcon { get; }
    public string? PublisherSlogan { get; }
    public IReadOnlyList<RoyalePublisherGame> PublisherGames { get; }
    public decimal Budget { get; }

    public decimal GetTotalFantasyPoints()
    {
        decimal? points = PublisherGames
            .Sum(x => x.FantasyPoints);
        return points ?? 0m;
    }
}

File: src/FantasyCritic.ClientAppVue2/src/models/royale/RoyalePublisher.ts
import { RoyaleYearQuarter } from './RoyaleYearQuarter.ts';
import { RoyalePublisherGame } from './RoyalePublisherGame.ts';
import { RoyaleAction } from './RoyaleAction.ts';
import { RoyalePublisherStatistics } from './RoyalePublisherStatistics.ts';

export interface RoyalePublisher {
  publisherID: string;
  yearQuarter: RoyaleYearQuarter;
  userID: string;
  playerName: string;
  publisherName: string;
  publisherIcon: string | null;
  publisherSlogan: string | null;
  publisherGames: RoyalePublisherGame[];
  publisherActions: RoyaleAction[];
  statistics: RoyalePublisherStatistics[];
  budget: number;
  totalFantasyPoints: number;
  ranking: number | null;
  quartersWon: RoyaleYearQuarter[];
  previousQuarterWinner: boolean;
  oneTimeWinner: boolean;
}
